0.15

balloob opened the pull request:
This will be the PR for release 0.15. I usually try to highlight one cool thing but this release actually has 4!
- @fabaff has upgraded the codebase to follow the PEP257 documentation standard.
- @partofthething has migrated us to use the main Python Open Z-Wave library instead of our forked version.
- To make our automations more powerful, @persandstrom added the option to use templates to dynamically create service calls. This works for automation, script, Alexa, universal media player, template switch.
- @MartinHjelmare has upgraded our scene support to now support all built-in services and components.

Besides bug fixes, this release also brings:
- Scene: Converted to a platform based component (@sander76)
- Scene: Hunter Douglas Powerview now supported (@sander76)
- Lock: MQTT platform added (@fabaff)
- Device Tracker: OwnTracks will allow filtering inaccurate GPS locations (@HydrelioxGitHub)
- Binary Sensor: Wemo Motion now supported (@pavoni, @ryanlaux)

``` yaml
# Example using templates for service and data in service call.
# Works for automation, script, Alexa, universal media player, template switch.
automation:
  - trigger:
      - platform: state
        entity_id: switch.bathroom
    action:
      service_template: >
        {% if is_state('switch.bathroom', 'on') %}
          switch.turn_on
        {% else %}
          switch.turn_off
        {% endif %}
      data_template:
        entity_id: switch.{{ states('input_select.is') }}
```

sander76 committed:
first commit

sander76 committed:
flake8 correction

sander76 committed:
- removed update method
- removed failing tests from test_scene

sander76 committed:
removed unnecessary properties.

balloob committed:
Version bump to 0.15.0.dev0

MartinHjelmare committed:
Fix mysensors sensor types

* Remove sprinkler and water leak from sensor types. These two are
  supported by binary sensor.

MartinHjelmare committed:
Merge pull request #1423 from MartinHjelmare/fix-mysensors-sensor-types

Fix mysensors sensor types

kk7ds committed:
Fix static configured wemo devices

The new wemo code was pulling 'static' from the global config instead of
the wemo component config.

kk7ds committed:
Fix wemo known device tracking

The wemo component was tracking devices by mac to avoid adding duplicates,
but this means we'd never be able to load more than one static wemo, since
they all have mac=None.

This changes the code to track by url, which has to be unique per wemo
anyway, and which we also have for even static wemos.

Commit:
Move hardcoded required Python version into homeassistant.const

balloob committed:
Merge pull request #1426 from shaftoe/python_constant

Move hardcoded required Python version into homeassistant.const

balloob committed:
Merge pull request #1425 from kk7ds/fix-static-wemos

Fix static configured wemo devices

balloob committed:
Fix incorrectly setting can_cancel on scripts

balloob committed:
Merge pull request #1429 from balloob/fix-script-can_cancel

Fix incorrectly setting can_cancel on scripts

balloob committed:
Update frontend

balloob committed:
Update PULL_REQUEST_TEMPLATE.md

sander76 committed:
1. added platform per scene entry.
2. changed homeassistant scene setup_platform method to work with 1.

sander76 committed:
removed logger.
I guess I should add some error checking, but I'd like you to okay this code first.

sander76 committed:
removed logger.
I guess I should add some error checking, but I'd like you to okay this code first.

fabaff committed:
Modify docstrings to match PEP257

ggruner committed:
Add support for NetAtmo rain gauge

Example Config:
...
modules:
    Bedroom:
      - temperature
    LivingRoom:
      - temperature
    Outside:
      - temperature
    Garden:
      - rain
      - sum_rain_1
      - sum_rain_24

The three new parameters stand for:
- rain # Estimated rainfall for today in "mm"
- sum_rain_1 # Rainfall in the last hour in "mm"
- sum_rain_24 # Rainfall in "mm" from 00:00am - 23:59pm

MartinHjelmare committed:
Add assumed_state to mysensors switch and light

* Return true for assumed_state if optimistic config setting is true.
* Fix PEP257

fabaff committed:
Add link to docs

fabaff committed:
Upgrade py-cpuinfo to 0.2.3

fabaff committed:
Upgrade to psutil 4.0.0

kk7ds committed:
Gracefully handle having no wemo config section

I broke this with my fix, where I assumed that we'd always have a wemo
config section if we're running the wemo code. However, if we're fully
auto-detected, we might not.

balloob committed:
Merge pull request #1431 from kk7ds/fix-wemo-fix

Gracefully handle having no wemo config section

balloob committed:
Merge pull request #1430 from ggruner/dev

Add support for NetAtmo rain gauge

fabaff committed:
Add MQTT lock support

sander76 committed:
Added your suggestions.
Looking at your code suggestion below I am not sure exactly how other people might want to put in lists. (But I am missing a more general overview of the code)

 ```
if not isinstance(scene_config,list):
  scene_config=[scene_config]
```

But it is there !
And changed "config" to "states" !

partofthething committed:
updated build script for unforking of python-openzwave

balloob committed:
Update netdisco to 0.5.3

nunofgs committed:
Add ArcherC7 150427 support

balloob committed:
Merge pull request #1436 from partofthething/zwave-unfork

updated build script for unforking of python-openzwave

balloob committed:
Docker + Z-Wave <3

balloob committed:
Merge pull request #1437 from balloob/update-docker-zwave

Docker + Z-Wave <3

balloob committed:
Update Dockerfile

balloob committed:
Update frontend

Commit:
Fixed the z-wave sensor workarounds for empty manufacturer ids

Commit:
Added tests for rfxtrx sensor

balloob committed:
Merge pull request #1440 from stefan-jonasson/fix-manufacturer-id

Fixed the z-wave sensor workarounds for empty manufacturer ids

kk7ds committed:
Give mFi options to control TLS usage

The default configuration of the mFi controller generates self-signed
certificates which are valid for short periods of time, and which are
regnerated on start or as needed. This makes requests mad. Since most
people will be using the self-signed certificates anyway, add options
to let them choose non-TLS, or unverified connections if they want/need.

This bumps the mficlient requirement to 0.3.0 for proper handling of
verify=False.

MartinHjelmare committed:
Merge pull request #1451 from MartinHjelmare/mysensors-assumed-state

Add assumed_state to mysensors switch and light

balloob committed:
Merge pull request #1450 from kk7ds/mfi-tls-verify

Give mFi options to control TLS usage

balloob committed:
Merge pull request #1433 from fabaff/mqtt-lock

Add MQTT lock support

balloob committed:
Merge pull request #1374 from sander76/scene_with_platforms

scene with platforms

balloob committed:
Update Dockerfile: link Z-Wave config

balloob committed:
Merge pull request #1452 from balloob/rfxtrx_tests

Added tests for rfxtrx sensor

kk7ds committed:
Track WeMo devices by serial number

This makes us track WeMo devices by serial number instead of URL. Per
@jaharkes' suggestion, tracking via URL could potentially be a problem
if a device changes to another IP and then appears to be discovered again.
Since we can't track based on mac (for static-configured devices), we
can use the serial number exposed to make sure we're always looking at
a list of unique devices.

jaharkes committed:
Update netdisco to 0.5.4

Commit:
Added tests for RFxtrx device, updated rfxtrx to ver 0.5, fixed bug in setting brightness for rfxtrx light

kk7ds committed:
Fix SNMP device_tracker results without MAC addresses

@llauren reported that his router returns empty MAC addresses for things
on different subnets, which causes heartache for the HA snmp device tracker.

balloob committed:
Merge pull request #1463 from kk7ds/snmp-track-macless-things

Fix SNMP device_tracker results without MAC addresses

balloob committed:
Merge pull request #1458 from jaharkes/track-wemos-by-serial

Track wemos by serial

bestlibre committed:
Possibility to blacklist events

Possibility to blacklist some events based on the entity_id

Commit:
update rfxtrx lib to ver 0.5

balloob committed:
Merge pull request #1466 from balloob/tests_rfxtrx_device

Tests rfxtrx device

Commit:
Update pychromecast 0.7.2

balloob committed:
Merge pull request #1471 from balloob/cast-update

Update pychromecast 0.7.2

pavoni committed:
Add Wemo Motion

HydrelioxGitHub committed:
Add GPS accuracy check for a location update or an event update
Add tests related to this
Great thanks to @pavoni for his support on this fix

pavoni committed:
Merge pull request #1474 from HydrelioxGitHub/Fix-Owntracks-GPS-Accuracy

Add GPS accuracy check for a location update or an event update

bpoirriez committed:
Use global variable and merge two if in one

balloob committed:
Merge pull request #1465 from bestlibre/influxdb_blacklist

Possibility to blacklist events for influxdb

Commit:
Add assumed state for Tellstick devices.
Tellstick device states are always assumed so this should be set to true.

balloob committed:
Merge pull request #1476 from stefan-jonasson/assumed-state

Add assumed state for Tellstick devices.

balloob committed:
TCP test to work offline

balloob committed:
Skip rfxtrx tests because of speed reasons

balloob committed:
History: Ignore insignificant domains

balloob committed:
History: ignore scripts that we cannot cancel

balloob committed:
Device tracker: remove support for old config

balloob committed:
Merge pull request #1434 from nunofgs/enhancement/add-archer-c7-support

Add ArcherC7 150427 support

balloob committed:
Merge pull request #1480 from balloob/fix-slow-tests

Fix slow tests

MartinHjelmare committed:
Refactor reproduce_state for scene component

* Add tests to reach full coverage for helpers/state.py.
* Refactor reproduce_state function in helpers/state.py. Add two dicts,
	as global constants, service_attributes and service_to_state. Use
	these in combination with the dict of services per domain from
	ServiceRegistry, to find the correct service to use in a scene state
	change.
* Use break statement in for loop, to break if service was selected
	to update state, in preference to update state attributes, ie state
	update takes precedence.
* Add ATTR_CODE and ATTR_CODE_FORMAT in const. Import these in
	alarm_control_panel and lock platforms instead of making duplicate
	constants in multiple modules.
* Use ATTR_MEDIA_CONTENT_TYPE and ATTR_MEDIA_CONTENT_ID in media_player
	platform in SERVICE_PLAY_MEDIA and play_media methods, instead of
	'media_type' and 'media_id'.
* Fix PEP257 in modified files.

balloob committed:
Group: fix thread safety

balloob committed:
Device Sun Light Trigger: Clean up

sander76 committed:
Powerview hub implementation

pavoni committed:
Force float for zone parameters.

pavoni committed:
Fix noisy error on startup. Make callbacks code consistent.

persandstrom committed:
After maintenance false password errors are expected

balloob committed:
Merge pull request #1491 from persandstrom/verisure_disable_password_lock

After maintenance false password errors are expected

balloob committed:
Merge pull request #1489 from balloob/zone_fix

Force zone params to be float

balloob committed:
Merge pull request #1488 from balloob/template_component_fix

Fix noisy error on template component startup.

balloob committed:
Merge pull request #1484 from balloob/device-tracker-remove-old-conf-format

Device tracker remove old conf format

balloob committed:
Merge pull request #1482 from balloob/history-significant-states

History: update significant states

balloob committed:
MQTT Light: fix brightness issues

fabaff committed:
Modify docstrings to match PEP257

fabaff committed:
Modify docstrings to match PEP257

balloob committed:
Fix TP-Link get auth token

fabaff committed:
Modify docstrings to match PEP257

fabaff committed:
Update docstrings to match PEP257

fabaff committed:
Update docstrings to match PEP257

balloob committed:
Merge pull request #1495 from balloob/mqtt-light

MQTT Light: fix brightness issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Revert change (FIXME)

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

balloob committed:
Merge branch 'master' into dev

Conflicts:
	homeassistant/components/wemo.py
	homeassistant/const.py

balloob committed:
Update frontend

balloob committed:
Merge pull request #1501 from fabaff/pep257-alarm

PEP257 - Alarm control panel

balloob committed:
Merge pull request #1503 from fabaff/pep257-automation

PEP257 - Automation

balloob committed:
Merge pull request #1505 from fabaff/pep257-binary-sensor

PEP257 - Binary sensor

balloob committed:
Merge pull request #1506 from fabaff/pep257-camera

PEP257 - Camera

balloob committed:
Merge pull request #1490 from sander76/powerview

Powerview

balloob committed:
Rename powerview to douglas_powerview

fabaff committed:
Merge branch 'pep257-device-tracker' into dev

fabaff committed:
Merge branch 'pep257-garage' into dev

fabaff committed:
Merge branch 'pep257-light' into dev

fabaff committed:
Merge branch 'pep257-lock' into dev

fabaff committed:
Merge branch 'pep257-scene-mqtt' into dev

fabaff committed:
Merge branch 'pep257-thermostats' into dev

fabaff committed:
Merge branch 'pep257-utils' into dev

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix PEP257 issues

sander76 committed:
change module name to correct brand name "douglas" to "hunterdouglas"

fabaff committed:
Fix PEP257 issues

persandstrom committed:
verisure configuration value typo fix

balloob committed:
Merge pull request #1507 from sander76/dev

change module name to correct brand name "douglas" to "hunterdouglas"

fabaff committed:
Fix PEPE257 issues

fabaff committed:
Merge branch 'pep257-sensor' into dev

fabaff committed:
Fix PEEP257 issues

fabaff committed:
Remove parenthesis

persandstrom committed:
Merge pull request #1508 from persandstrom/verisure_typo_thermometers

verisure configuration value typo fix

fabaff committed:
Fix PEP257 issues

fabaff committed:
Merge branch 'pep257-tests' into dev

fabaff committed:
Fix PEP257 issues

fabaff committed:
Fix docstring

balloob committed:
Merge pull request #1485 from MartinHjelmare/refactor-scene-reproduce_state

Refactor reproduce_state for scene component

balloob committed:
Add pep257 checking

balloob committed:
Fix final pep257 issues

balloob committed:
Add pydocstyle to requirements_test.txt

fabaff committed:
Merge pull request #1513 from balloob/pep257

Enable Pep257 checking on CI

pavoni committed:
Merge pull request #1473 from balloob/add_wemo_motion

Add Wemo Motion device as a binary sensor.

fabaff committed:
Fix formatting

sander76 committed:
changed the powerview version

fabaff committed:
Add sensor_class to MQTT binary sensor

persandstrom committed:
Use templates in service calls

balloob committed:
Type checks for MQTT config

sander76 commented:
@balloob Do you want me to update the docs for the change hass scene and powerview scene ? Or is this done automatically ? (taking info from the PR)

Bart274 committed:
add gps_accuracy and battery information to the see function

the see function from device_tracker doesn't pass the gps_accuracy and the battery information to the see service, so this information gets lost.

Bart274 committed:
Update __init__.py

balloob commented:
That's not done automatically, pr welcome

On Fri, Mar 11, 2016, 04:09 sander76 <notifications@github.com> wrote:

> @balloob <https://github.com/balloob> Do you want me to update the docs
> for the change hass scene and powerview scene ? Or is this done
> automatically ? (taking info from the PR)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/1502#issuecomment-195342364>
> .
>

turbokongen committed:
Workaround for no off motion sensor: Wenzhou tkd slim sensor 3/4 in 1 sensor

fabaff committed:
Upgrade limitlessled to 1.1.0

fabaff committed:
Revert "Upgrade limitlessled to 1.1.0"

This reverts commit 2e2537de2a51fd3054975e3c0025f28deede6aab.

balloob committed:
Merge pull request #1523 from Bart274/patch-1

add gps_accuracy and battery information to the see function

balloob committed:
Merge pull request #1519 from sander76/dev

changed powerview version requirement to solve error in installing

balloob committed:
Merge pull request #1520 from balloob/mqtt-fixes

Type checks for MQTT config

balloob committed:
Merge pull request #1524 from turbokongen/dev

Workaround for no off motion sensor: Wenzhou tkd slim sensor 3/4 in 1…

balloob committed:
Merge pull request #1510 from persandstrom/automation_templating

Use templates in service calls

balloob committed:
Update .coveragerc

balloob committed:
Merge pull request #1457 from fabaff/sensor-classes

Add support for sensor classes

balloob committed:
Update frontend

balloob committed:
Update frontend build scripts for pep257

balloob committed:
Version bump to 0.15
